mux/detector.go

```diff
@@ -0,0 +1,87 @@
+package mux
+
+import (
+	"net/http"
+
+	botdetector "github.com/devopsfaith/krakend-botdetector"
+	krakend "github.com/devopsfaith/krakend-botdetector/krakend"
+	"github.com/luraproject/lura/config"
+	"github.com/luraproject/lura/logging"
+	"github.com/luraproject/lura/proxy"
+	"github.com/luraproject/lura/router/mux"
+	luramux "github.com/luraproject/lura/router/mux"
+)
+
+// New checks the configuration and, if required, wraps the handler factory with a bot detector middleware
+func New(hf luramux.HandlerFactory, l logging.Logger) luramux.HandlerFactory {
+	return func(cfg *config.EndpointConfig, p proxy.Proxy) http.HandlerFunc {
+		next := hf(cfg, p)
+
+		detectorCfg, err := krakend.ParseConfig(cfg.ExtraConfig)
+		if err == krakend.ErrNoConfig {
+			l.Debug("botdetector: ", err.Error())
+			return next
+		}
+
+		if err != nil {
+			l.Warning("botdetector: ", err.Error())
+			return next
+		}
+
+		d, err := botdetector.New(detectorCfg)
+		if err != nil {
+			l.Warning("botdetector: unable to create the LRU detector:", err.Error())
+			return next
+		}
+
+		return handler(d, next)
+	}
+}
+
+func handler(f botdetector.DetectorFunc, next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if f(r) {
+			w.WriteHeader(http.StatusForbidden)
+			w.Write([]byte("bot rejected"))
+			return
+		}
+		next(w, r)
+	}
+}
+
+type middleware struct {
+	detector botdetector.DetectorFunc
+}
+
+// Register checks the configuration and, if required, registers a bot detector middleware at the gin engine
+func Register(cfg config.ExtraConfig, l logging.Logger) mux.HandlerMiddleware {
```

Comment:
If no engine can be provided to the `Register` function, I think we should change the name of the function (maybe to `NewMiddleware`?)

Reply:
8523715484064e3aeb822cb5161c24c72d4f233f, I renamed it NewMiddleware, at first I thought so but I didn't want to lose the format.

But it makes a lot more sense to launch it with NewMiddleware.